### PR TITLE
Java major mode

### PR DESCRIPTION
**Pull request by erichonorez**
Hello,

This is my proposal for the java major mode (Ref #98).

### CODE REVIEW

**Comment by MarcoIeni** [comment]
Thanks for this!
Is installing any additional extension required?

File: package.json [hunk 1]
```diff
@@ -7985,5 +8323,16 @@
...
+    "__metadata": {
+        "id": "1c81ab96-0424-43c4-b356-fe408a1bd1cf",
+        "publisherId": "60415ab6-4581-4e73-a7e0-6fc6b3369f12",
+        "publisherDisplayName": "VSpaceCode",
+        "targetPlatform": "undefined",
+        "isApplicationScoped": false,
+        "updated": false,
+        "isPreReleaseVersion": false,
+        "installedTimestamp": 1672343009383,
+        "preRelease": false
```
Comment (MarcoIeni) [inline package.json:8336]: what's this `__metadata` field?

Comment (erichonorez) [reply package.json:8336]: I didn't notice it.
It has probably been added when I packaged a vsix locally.

I'm going to remove these lines in a new commit.

**Comment by erichonorez** [comment]
> Thanks for this! Is installing any additional extension required?

Of course you need to have the `Language Support for Java`, `Debugger for Java` and `Test runner for Java` extensions installed (all found in the `Extension Pack for Java` bundle).

File: package.json (hunk 1 above)
Comment (erichonorez) [reply package.json]: It has been removed in the last commit (490c49bc849364aed8e4facaa0bf84aff911bdd5)

**Comment by MarcoIeni** [comment]
> > Thanks for this! Is installing any additional extension required?
> 
> Of course you need to have the `Language Support for Java`, `Debugger for Java` and `Test runner for Java` extensions installed (all found in the `Extension Pack for Java` bundle).

Great, can you make a PR to update [this](https://github.com/VSpaceCode/vspacecode.github.io/blob/master/docs/major-mode.md) page in the docs?

**Comment by MarcoIeni** [comment]
I noticed you are missing `spc = S` from the [conventions](https://vspacecode.github.io/docs/conventions#common-major-mode-key-bindings). Can you make sure you have all the suggested key bindings from the list?

File: package.json [hunk 2]
```diff
@@ -6697,6 +6697,379 @@
...
+                                                "icon": "zap",
+                                                "type": "bindings",
+                                                "bindings": [
+                                                    {
+                                                        "key": "/",
+                                                        "name": "Comment line",
+                                                        "icon": "edit",
+                                                        "type": "command",
+                                                        "command": "editor.action.commentLine"
+                                                    },
```
Comment (MarcoIeni) [inline package.json:6789]:
```suggestion
```
This is already available with `spc ;`, which is faster to type :)

File: package.json [hunk 3]
```diff
@@ -6697,6 +6697,379 @@
...
+                                                    {
+                                                        "key": "e",
+                                                        "name": "Extract to function or variable",
+                                                        "icon": "gather",
+                                                        "type": "command",
+                                                        "command": "editor.action.codeAction",
+                                                        "args": {
+                                                            "kind": "refactor.extract"
+                                                        }
```
Comment (MarcoIeni) [inline package.json:6975]: can you move this up, so that the file is alphabetically ordered?

**Review by MarcoIeni — CHANGES_REQUESTED** [review]
looks good. Just some minor comments.

File: package.json (hunk 3 above)
Comment (erichonorez) [reply package.json]: Done

File: package.json (hunk 2 above)
Comment (erichonorez) [reply package.json]: Removed

**Review by MarcoIeni — APPROVED** [review]
Thanks!

**Pull request merged**